### PR TITLE
Report exit code of rsync processes if they fail in TestWithRsync

### PR DESCRIPTION
**Pull request by capnspacehook**
While at it fix incorrect use of `require` assertions inside of a `require.Eventually` call.

### CODE REVIEW

File: tool/tsh/common/proxy_test.go [hunk 1]
```diff
@@ -477,7 +481,12 @@ func TestWithRsync(t *testing.T) {
 			cmd.Stdout = os.Stdout
 			cmd.Stderr = os.Stderr
 			err = cmd.Run()
-			require.NoError(t, err)
+			if !assert.NoError(t, err) {
+				var exitErr *exec.ExitError
+				if errors.As(err, &exitErr) {
+					t.Logf("exit code: %d", exitErr.ExitCode())
+				}
+			}
```
Comment (zmb3) [inline tool/tsh/common/proxy_test.go:489]: Would be nice if the exit code was included in the test failure as opposed to showing up later in the logs. What about something like this?
```suggestion
			var msg string, exitErr *exec.ExitError
			if errors.As(err, &exitErr) {
				msg = fmt.Sprintf("exit code: %d", exitErr.ExitCode()
			}
			require.NoError(t, err, msg)
```

Comment (capnspacehook) [reply tool/tsh/common/proxy_test.go]: Good idea thanks

**Review by lxea — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Comment by capnspacehook** [comment]
@zmb3 the flaky test detector failed due to a time-out, can I get that check ignored please?

**Pull request merged**